src/mlpack/methods/ann/layer/mean_pooling_impl.hpp

```diff
@@ -83,8 +123,31 @@ void MeanPooling<InputDataType, OutputDataType>::Forward(
   outputTemp = arma::zeros<arma::Cube<eT> >(outputWidth, outputHeight,
       batchSize * inSize);
 
-  for (size_t s = 0; s < inputTemp.n_slices; s++)
+  bool isPadded {padWLeft != 0 || padWRight != 0 ||
+      padHTop != 0 || padHBottom != 0};
+
+  if (isPadded)
```

Comment:
I think you could combine the if conditions in line 129 and 149 as they have the same condition.